socs/agents/ibootbar/agent.py

```diff
@@ -414,6 +428,9 @@ def set_initial_state(self, session, params=None):
         Performs a software reboot. The outlets are then set to their
         respective initial states. This takes about 30 seconds.
         """
+        if self.ibootbar_type == 'IBOOTBAR':
+            return True, 'Skipped because this is not supported for IBOOTBAR.'
```

Comment:
I'd probably opt to fail out here by returning `False`. I don't think this is a commonly used task, but if someone is using it expecting it to work and checking the return status, they might miss that this is really a no-op.

```suggestion
            return False, 'Software reboot is not supported for IBOOTBAR type devices.'
```